### ISSUE TITLE
Manually adding errors to a property marks it as no longer dirty

### ISSUE THREAD

**Issue by wmadden** [poteto/ember-changeset#177]
https://github.com/DockYard/ember-changeset/blame/master/addon/index.js#L356

When you call `pushErrors(key, ...)` it appears to deliberately clear the changes on that key. There's no explanation why this is, and it's problematic when you're adding server-side errors to the changeset because after calling `pushErrors()` all previously-changed but erroneous properties are no longer dirty. If there were no other changes on the changeset, `isDirty()` now returns `false`.

Is this behaviour intentional? If so why? If not, this is a bug.

**Comment by phcoliveira** [comment]
It appears that manually adding errors does not get its consumers notified. I have a standard use case of an error message below an input. All client-side validations are handled perfectly, so it is to be expected that, by following the instructions for server-side on README, the error message returned by the server would also appear.

**Comment by shankarsridhar** [comment]
I have the same problem. While adding errors manually using `pushErrors` or `addError`, I don't see the changeset's errors updated in the UI although the data when logged in the console, shows up just fine.

**Comment by mhluska** [comment]
For those coming here wondering how to hack around this issue, you just need to add `changeset.notifyPropertyChange('_errors');` after the server error updates:

```
changeset.save()
  .then(() => { /* ... */ })
  .catch(() => {
    get(this, 'model.errors').forEach(({ attribute, message }) => {
      changeset.pushErrors(attribute, message);
    });
    changeset.notifyPropertyChange('_errors');
  });
```

**Comment by snewcomer** [comment]
I see this is a potential bug.  Here is a PR #311 to fix that.  Regarding the notifications, I believe we are "notifying" correctly (at least on the latest master).  Lmk if anybody sees anything differently 👍

**Comment by mhluska** [comment]
@snewcomer the PR didn't fix the issue for me in v1.5.0. I still have to do my `changeset.notifyPropertyChange('_errors');` to get server errors to appear.

**Comment by mhluska** [comment]
@snewcomer the PR doesn't seem to fix the issue for me on v1.5.0. I still have to use my `changeset.notifyPropertyChange('_errors');` hack above to get server errors to appear. Granted it could be a different issue.
